### DISCUSSION TITLE
new DB/filesystem layout idea

### DISCUSSION THREAD

**Discussion by holgerBerger** [holgerBerger/hpc-workspace-v2#3] [Ideas]
There is a wishes for better separation of user groups on large systems.

One idea is to work with stricter permission sets, and deeper less accessible directory structures
to hide more information about existing workspaces from others, and make information leaks (nearly) impossible.

One idea is a path structure like this:

/scratch/ws/<org>/<group>/<user>-<wsname>

where _org_ is a virtual organization, that is a set of unix groups. Think of it like a institute or a company.
Data sharing is possible and easy within an _org_, as it is within a group (already now).

```
/scratch						root:root 	drwxr-xr-x
	 /ws						root:root 	drwxr-xr-x
	    /<org>					ws:ws 	        drwx--x--x
		    /<group>			        ws:group	drwx--x---
				/<user>-<wsname> 	user:group	drwx------

```
for sharing there would be a special directory
```
/scratch/ws/<org>/shared/<user>-<wsname>           
```
with permissions d-----x--x  and ACLs (this works only if ACLs are supported) set by ws_allocate to allow specific users
and groups (of same _org_!?) to access the workspace.

Opinions?
Is that useful?
Better ideas?

**Comment by cniethammer** [comment]
After reading this I would start asking the following potential design-relevant questions:
- Are there more organisational layers intended for the future?
- What grade of separation shall be achieved exactly? (strictly prohibit sharing outside of organisation/group - even if ACLs are set wrong in some places?)
- are read-only shares ok, or is write access mandatory?
- why not super simple
```

ws: special workspace user + ws group with nobody inside except the ws user
<user>: user / individual person

/scratch [root:root] drwx--x--x
          /ws    [ws:ws] drwx--x--x
                          /ws-id [<user>:ws] drwx------ + ACLs for everything else?
```

- reliability /resilience needs of the system: if one filesystem is offline, data should be safe and other filesystems sill be accessible.
